### ISSUE TITLE
Get distance on basis of route calculation. 

### ISSUE THREAD

**Issue by Anas-jed** [KohlsAdrian/google_maps_utils#9]
<details>
<summary>Flutter Doctor</summary>

```
[√] Flutter (Channel stable, 2.5.3, on Microsoft Windows [Version 10.0.19043.1288], locale en-US)
    • Flutter version 2.5.3 at C:\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 18116933e7 (13 days ago), 2021-10-15 10:46:35 -0700
    • Engine revision d3ea636dc5
    • Dart version 2.14.4

[√] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at C:\Users\AnasYousuf\AppData\Local\Android\sdk
    • Platform android-31, build-tools 31.0.0
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Android Studio (version 2020.3)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[√] VS Code (version 1.60.2)
    • VS Code at C:\Users\AnasYousuf\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.26.0

[√] Connected device (2 available)
    • Chrome (web) • chrome • web-javascript • Google Chrome 95.0.4638.54
    • Edge (web)   • edge   • web-javascript • Microsoft Edge 95.0.1020.30

• No issues found!
```

</details>

the distance is calculated with direct line.

what i need is to give me distance on the basis of route (means it takes road as a calculation parameter) not with direct straight line. so, if my requirement is included in this package kindly let me know.

attaching images for better understanding.

![p1](https://user-images.githubusercontent.com/43311899/139247120-0d7dca1c-18b8-48d1-be00-63d5f8d6052d.jpg)
![p2](https://user-images.githubusercontent.com/43311899/139247127-b61da978-df93-459b-a8ef-1a6b3718d176.jpg)

**Comment by KohlsAdrian** [comment]
For the calculation to work, you should have a list of points, I can implement a function that can make this work, but this is kinda easy to solve without the package, you can calculate distance of two points iterating in the whole list of points and sum every tuple and at the end you have the result

**Comment by KohlsAdrian** [comment]
is this easy to implement, but I am adding to the package anyway

``` dart
double total = 0.0;

    for (int i = 0; i < points.length - 1; i++) {
      total += computeDistanceBetween(points[i], points[i + 1]);
    }

    return total;

```

**Comment by KohlsAdrian** [comment]
published as 1.5.0 on pub.dev
